src/olympia/promoted/views.py

```diff
@@ -0,0 +1,35 @@
+import stripe
+
+from django.http import HttpResponse
+from django.views.decorators.csrf import csrf_exempt
+
+import olympia.core.logger
+
+from olympia.amo.decorators import post_required
+
+from .utils import create_stripe_webhook_event
+
+
+log = olympia.core.logger.getLogger("z.promoted")
+
+
+@post_required
+@csrf_exempt
+def stripe_webhook(request):
+    payload = request.body
+    sig_header = request.META.get("HTTP_STRIPE_SIGNATURE")
+
+    try:
+        event = create_stripe_webhook_event(
+            payload=payload, sig_header=sig_header
+        )
+    except stripe.error.SignatureVerificationError:
+        log.exception("received stripe event with invalid signature")
+        return HttpResponse(status=400)
+    except ValueError:
+        log.exception("received stripe event with invalid payload")
+        return HttpResponse(status=400)
+
+    log.info('Received stripe event type: "%s".', event.type)
```

Comment:
so we're not actually doing anything with the event at this time?  Would the functionality sit in `create_stripe_webhook_event`?  I'm wondering how the processing flow is going to work here.

Reply:
My idea would be to submit a Celery task, which we would have to create for each event we want to listen to and handle. This only receives and validates notifications and once we have an `event` object, we can do whatever we want (which is the complicated part, heh).

We'll see shortly with payment failures, which we should handle by sending an email to amo admins.